scene/3d/xr_nodes.cpp

```diff
@@ -346,17 +342,22 @@ void XRNode3D::_bind_tracker() {
 		}
 
 		tracker->connect("pose_changed", callable_mp(this, &XRNode3D::_pose_changed));
+		tracker->connect("pose_lost_tracking", callable_mp(this, &XRNode3D::_pose_lost_tracking));
 
 		Ref<XRPose> pose = get_pose();
 		if (pose.is_valid()) {
 			set_transform(pose->get_adjusted_transform());
+			_set_has_tracking_data(pose->get_has_tracking_data());
+		} else {
+			_set_has_tracking_data(false);
 		}
 	}
 }
 
 void XRNode3D::_unbind_tracker() {
```

Comment:
Should `_unbind_tracker()` also set `has_tracking_data = false`?

If not, I think that could mean that `get_has_tracking_data()` could continue to return `true` even after the tracker or pose has gone away. However, I didn't actually test this.

Reply:
What I was trying to do was prevent unnecessary `tracking_changed` signals when switching trackers. I analyzed when the `_bind_tracker()` and `_unbind_tracker()` methods were called and made sure that the signal would remain accurate. For example an `_unbind_tracker()` without a subsequent `_bind_tracker()` only occurs in the `_removed_tracker()` signal handler, so in that case I manually added a `_set_has_tracking_data(false)` call.

On further consideration I'm wondering whether this is over-engineering. If a node really switches from one tracker to another then I see no problem with `tracking_changed(false)` being emitted at the old location, and then `tracking_changed(true)` being emitted at the new location. As such I'll just move the clearing to be unconditional in `_unbind_tracker()`.

Reply:
I think so, changing trackers is a very unlikely scenerio so having an extra signal broadcast isn't a big issue I think.

Generally speaking these things are set on the node once. Your left hand isn't suddenly going to change into your right hand.

The real issue is that when a tracker is set, the tracker may not exist yet, it may become available much later in the lifetime of a game. 
That said, that was mostly true for OpenVR, in OpenXR we create trackers at initialisation for all paths, but keep them inactive if nothing is actually bound to the input paths.

Reply:
I think XRNode3D cleanly handles the trackers coming on line.
- If XRNode3D is set to a tracker before it exists then `XRNode3D::set_tracker` saves the tracker name, and when XRServer sends `tracker_added` the XRNode3D will bind to the tracker and set `has_tracking_data` true (if the pose is valid).
- If XRNode3D is set to a tracker before the first pose is available then when the XRPositionalTracker sends the first `pose_changed` the XRNode3D will then set `has_tracking_data` true.